### PR TITLE
[VI] Update `Contribution_guide/Best-practices`

### PR DESCRIPTION
**Pull request by TranT3011**
rewriting most of the things based on https://clayton.cc/osu-wiki-status/diff-vi-wiki-osu!_wiki-Contribution_guide-Best_practices (changes since the vietnamese translation was last updated which was 24th of september 2021)

### CODE REVIEW

File: wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md [hunk 1]
```diff
@@ -4,61 +4,31 @@ outdated_translation: true
 outdated_since: 1963510e21506f90b21ad201b0eb9ca98fe7ecb5
 ---
 
-# Các vấn đề hay gặp của người đóng góp wiki
+# Các gợi ý để làm công việc của bạn tốt hơn
 
-## Ai đó bảo tôi hãy viết thông điệp commit có ý nghĩa!
+## Bài viết này sẽ nói về một số khó khăn có thể bạn đối mặt khi đang đóng góp. Các phương pháp được đề cập ở đây là làm cho công việc dễ hơn và có thể được áp dụng cho các dự án khác được quản lý trên GitHub hoặc các phương tiện tương tự.
```
Comment (TPGPL) [inline wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md:3]:
```suggestion
Bài viết này sẽ nói về một số khó khăn có thể bạn đối mặt khi đang đóng góp. Các phương pháp được đề cập ở đây là làm cho công việc dễ hơn và có thể được áp dụng cho các dự án khác được quản lý trên GitHub hoặc các phương tiện tương tự.
```

i think this is supposed to be a paragraph, not a header

File: wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md [hunk 2]
```diff
@@ -4,61 +4,31 @@ outdated_translation: true
...
 
-Điều này cũng có thể diễn giải rằng "Mọi commit của tôi có thông điệp `Update vi.md` hay gì đó tương tự".
+## Mở đầu.
```
Comment (TPGPL) [inline wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md:5]:
```suggestion
## Mở đầu
```

File: wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md [hunk 3]
```diff
@@ -4,61 +4,31 @@ outdated_translation: true
...
 
-Nếu ai đó bảo bạn điều này hoặc bạn thấy mọi thông điệp commit đều giống nhau, điều đó có nghĩa là bạn không để lại thông điệp commit có ý nghĩa khi bạn di chuyển, sửa đổi, và/hoặc xoá các tệp trên Giao diện Web GitHub.
+*Để biết thêm thông tin về Git và GitHub, hãy xem [Tài liệu GitHub](https://docs.github.com)*.
```
Comment (TPGPL) [inline wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md:7]:
```suggestion
*Để biết thêm thông tin về Git và GitHub, hãy xem [Tài liệu GitHub](https://docs.github.com).*
```

File: wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md [hunk 4]
```diff
@@ -78,22 +48,55 @@ Nếu bạn gặp vấn đề gì khi đang sử dụng công cụ GitHub hoặc
 
 4. Nhấn nút `Run Workflow` và chờ nó hoàn thành. Nếu bạn tò mò xem công cụ này hoạt động ra sao, nhấn lại vào `Sync from osu! upstream`.
```
Comment (TPGPL) [inline wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md:43]: missing an image

for reference:
```md
4. Click the `Run Workflow` button and wait for the workflow to complete. If you're curious about how the tool works, click on the `Sync from osu! upstream` workflow task.

![](img/github-actions-workflow-overview.png "GitHub Actions Workflow - Workflow Overview")
```

File: wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md [hunk 5]
```diff
@@ -78,22 +42,55 @@ Nếu bạn gặp vấn đề gì khi đang sử dụng công cụ GitHub hoặc
...
 
-![GitHub Actions Workflow - Tổng quát về Workflow](img/github-actions-workflow-overview.png "GitHub Actions Workflow - Tổng quát về Workflow")
+## Chỉnh sửa
+
+*Xem thêm: [Forking Workflow | Hướng dẫn Atlassian Git](https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow)*
+
+Trong phạm vi bản fork của bạn, bạn có thể tuỳ ý chỉnh sửa và lưu chúng. **Commits** là từng "phiên bản" của repo. **Branches** là không gian làm việc, bạn có thể chuyển đổi các phiên bản của repo. Để làm quá trình làm việc dễ dàng hơn và làm cho lịch sử của wiki sạch và không lộn xộn, lưu ý các điều sau:
+- [Đồng bộ branch `master`](#syncing-the-fork).
```
Comment (TPGPL) [inline wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md:50]:
```suggestion
Trong phạm vi bản fork của bạn, bạn có thể tuỳ ý chỉnh sửa và lưu chúng. **Commits** là từng "phiên bản" của repo. **Branches** là không gian làm việc, bạn có thể chuyển đổi các phiên bản của repo. Để làm quá trình làm việc dễ dàng hơn và làm cho lịch sử của wiki sạch và không lộn xộn, lưu ý các điều sau:

- [Đồng bộ branch `master`](#đồng-bộ-fork).
```

File: wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md [hunk 6]
```diff
@@ -78,22 +42,55 @@ Nếu bạn gặp vấn đề gì khi đang sử dụng công cụ GitHub hoặc
...
+  - [Tự động đóng issue/pull request](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
+- Hãy nhớ chọn `Allow edits from maintainers`, nó sẽ để những người quản lý wiki giúp bạn cải thiện pull request nếu cần thiết
+
+## Sửa sót theo đánh giá
+
+Cách tốt nhất để sửa theo đánh giá là qua giao diện web của GitHub. Bấm `Add suggestion to batch` khi trong `Files changed` tab để sửa theo nhiều đánh giá cùng một lúc.
+
+![Sửa theo nhiều đánh giá](img/applying-batch-review.gif)
+
+Bạn có thể bấm `Commit suggestion` để sửa từng gợi ý một, nếu như bạn không viết linh tinh chả vì lý do gì và [với lời nhắc nhở có ý nghĩa](#making-edits).
```
Comment (TPGPL) [inline wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md:74]:
```suggestion
Bạn có thể bấm `Commit suggestion` để sửa từng gợi ý một, nếu như bạn không viết linh tinh chả vì lý do gì và [với lời nhắc nhở có ý nghĩa](#chỉnh-sửa).
```

File: wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md [hunk 7]
```diff
@@ -78,22 +42,55 @@ Nếu bạn gặp vấn đề gì khi đang sử dụng công cụ GitHub hoặc
...
-   3. Từ đây, sửa các conflict thủ công và xoá các đánh dấu `<<<<<<<`, `=======`, và `>>>>>>> master`.
+   3. Từ đây, sửa các conflict thủ công và xoá cả dòng có các đánh dấu `<<<<<<<`, `=======`, và `>>>>>>> master`.
    4. Lặp lại quá trình trên với mọi conflict.
    5. Khi hoàn thành, nhấn `Mark as resolved` (bạn chỉ có thể làm điều này khi mọi phần conflict trong tệp đã được xử lý).
-2. Nếu nút `Resolve conflicts` bị khoá vì nó quá phức tạp đối với GitHub, bạn đã gặp xui xẻo và cần phải [cập nhật branch của bạn](#branch-của-tôi-đã-lỗi-thời!) và thực hiện lại các thay đổi của bạn.
-   - *Chú ý: Điều này chỉ đúng khi bạn giới hạn ở việc sử dụng Giao diện Web GitHub.*  Vẫn còn nhiều cách khác để sửa, nhưng chúng sẽ không được đề cập trong bài viết này, và có thể nó không đáng để làm vậy, bởi vì đằng nào bạn cũng sẽ phải ghi đè và hoàn tác lại các thay đổi bị conflict.
+2. Nếu nút `Resolve conflicts` bị khoá vì nó quá phức tạp đối với GitHub, bạn đã gặp xui xẻo và cần phải [cập nhật branch của bạn](#syncing-the-fork) và thực hiện lại các thay đổi của bạn.
```
Comment (TPGPL) [inline wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md:95]:
```suggestion
2. Nếu nút `Resolve conflicts` bị khoá vì nó quá phức tạp đối với GitHub, bạn đã gặp xui xẻo và cần phải [cập nhật branch của bạn](#đồng-bộ-fork) và thực hiện lại các thay đổi của bạn.
```

**Review by TPGPL — CHANGES_REQUESTED** [review]

File: wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md (hunk 4 above)
Comment (TranT3011) [reply wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md:43]: i cant put an alternative text sit next to an image?

Comment (TPGPL) [reply wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md:43]: i think i would follow the English article, and not put the alt-text to that image

Comment (TranT3011) [reply wiki/osu!_wiki/Contribution_guide/Best_practices/vi.md:43]: ok i was a little stupid so i thought i was messing the other stuff
added

**Review by TPGPL — APPROVED** [review]

**Pull request merged**